### PR TITLE
GeoSay

### PR DESCRIPTION
**Pull request by CharlesAuthier**

### CODE REVIEW

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Thus, the included angles $$\theta$$ of L-junctions will also have special distribution when they are located along buildings.
+To verify this supposition, they calculated the distributions of L-junctions included angles among different regions from the **Spacenet65** dataset.
+One can find that junctions included angles are really close to $$\pi/2$$ in the building area and have a large difference towards junctions among background area.
+In building area, angles of junctions are highly concentrated in interval [$$\pi / 3$$, $$2\pi / 3$$].
+While in the background area, distribution around $$\pi/2$$ has less contrast to other intervals.
+Such distributions can help us to distinguish junctions around buildings from other objects, and can be used as a *prior* in the detection of buildings.
+
+![](/deep-learning/images/GeoSay/angle.png)
+
+#**GeoSay: from junctions to building index**
```
Comment (lemairecarl) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: Put a space after the #

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+Given an image, the geometric saliency is derived from a mid-level geometric representations based on the junctions that can locally describe geometrical structures of images.
+Additionally, GBI preserves both the exact position and accurate shape of single buildings compared to existing methods.
+
+![](/deep-learning/images/GeoSay/gbi.png)
+
+## Method
+
+# **Junction-based representation of images**
+Detecting junctions in an image is to find all the local structures *j*, modeled by the template and estimate their parameters.
+They used the a-contrariojunction detector (ACJ) where they assumed that the scale of branches are identical.
```
Comment (lemairecarl) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: "a-contrario junction" ? (space)

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+#**GeoSay: from junctions to building index**
+
+Given an image, all detected junctions *J* can be divided into two subsets,i.e., $$J=J_{B}\cup J_{\bar{B}}$$, where $$J_{B}$$ mean inside the buildings and $$J_{\bar{B}}$$ outside the buildings.
+For a junction *j* with its parametric description $$\theta_{j}=\{c, \vec{\nu}_{1}, \vec{\nu}_{2}, \beta, \rho\}$$, the posterior probability $$P(j\in J_{B}|\theta_{j})$$, measuring the possibility of the event that ajunction *j* is inside buildings, can be derived by
+
+![](/deep-learning/images/GeoSay/eq7.png)
+
+Where the prior probabilities $$P(J_{B})$$, $$P(J_{\bar{B}})$$ and the likelihoods $$P(\theta_{j}\|J_{B})$$, $$P(\theta_{j}\|J_{\bar{B}})$$ can be estimated from a given dataset of buildings, e.g. the Spacenet65 dataset, based on the fitted GMM model.
+
+By combining the significance parameter and included angle of a single junction, the first-ordergeometric saliency $$g^{(1)}_{j}$$ of a junction *j* can be computed as
```
Comment (lemairecarl) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: "first-order geometric" (space)

Smells like copy paste spirit

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+![](/deep-learning/images/GeoSay/eq7-2.png)
+
+Which indicates the degree of a single junction locating along buildings.
+
+![](/deep-learning/images/GeoSay/paire.png)
+
+An example of junctions andneighboring junctions is displayed in Fig.5, where green points are the centers of junctions inside the $$\tho-NN$$ of the junction with location center in red.
```
Comment (lemairecarl) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: andneighboring

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+Based on neighboring junctions, the pair-wise geometric saliency of a junction $$j$$ is defined as
+
+![](/deep-learning/images/GeoSay/eq9.png)
+
+**Geometric building index**, attempts to associate each pixelpwith a saliency measuring the possibility of the pixel belonging to buildings, which is the summationof saliency inside parallelogram of all junctions.
```
Comment (lemairecarl) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: pixelpwith
summationof

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
+---
+layout: review
+title:  "GeoSay: A Geometric Saliency for Extracting Buildings in Remote Sensing Images"
+tags:   deep-learning, segmentation, remote-sensing
```
Comment (jhlegarreta) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: We don't need (must not use?) commas to separate tags.

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+author: Charles Authier
+pdf:  https://arxiv.org/abs/1811.02793
+cite:
+  authors: "Gui-Song Xia, Jin Huang, Nan Xue, Qikai Lu, Xiaoxiang Zhu"
+  title:   "GeoSay: A Geometric Saliency for Extracting Buildings in Remote Sensing Images"
+  venue:   "7 nv 2018	arXiv:1811.02793 "
+---
+
+The code of ASJ: https://github.com/cherubicXN/anisotropic-scale-junction-detector
+The model of HF-FCN: https://github.com/tczuo/HF-FCN-for-Robust-Building-Extraction
```
Comment (jodoin) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: put * to make these 3 lines look like bullets.

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
+---
+layout: review
+title:  "GeoSay: A Geometric Saliency for Extracting Buildings in Remote Sensing Images"
+tags:   deep-learning segmentation remote-sensing
```
Comment (jodoin) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: is it really a deep-learning paper?

File: deep-learning/_posts/2019-05-02-GeoSay.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+# **Junction-based representation of images**
+Detecting junctions in an image is to find all the local structures *j*, modeled by the template and estimate their parameters.
+They used the a-contrario junction detector (ACJ) where they assumed that the scale of branches are identical.
+With ACJ detector, from the intersected point of a junction, they need to define a measurement to judge whether there exists junction or not and find those branches.
+(ASJ: Anisotropic Scale Junction Detection and Matching)
+
+![](/deep-learning/images/GeoSay/asj.png)
+
+# **Statistics of junctions in VHR RS images**
+One of the observations made is that, in RS images, the statistics of junctions should different on buildings and background.
```
Comment (jodoin) [inline deep-learning/_posts/2019-05-02-GeoSay.md]: should be

File: deep-learning/_posts/2019-05-02-GeoSay.md (hunk 6 above)
Comment (jhlegarreta) [reply deep-learning/_posts/2019-05-02-GeoSay.md]: Addressed in 2027834.

**Pull request merged**